### PR TITLE
static_website_tls with certs and http -> https redirect

### PR DESCRIPTION
**Pull request by azul**
Created based on mirage-seal.
Removed the configuration options and using functoria.

This is an alternative to #120.
Both solve #119.

Keys were taken from ocaml-tls repo.

### CODE REVIEW

**Comment by samoht** [comment]
That looks great, thanks!

File: static_website_tls/dispatch.ml [hunk 1]
```diff
@@ -1,56 +1,106 @@
...
+
+  module TCP  = S.TCPV4
+  module TLS  = Tls_mirage.Make (TCP)
+  module X509 = Tls_mirage.X509 (KEYS) (Clock)
+
+  module Http  = Cohttp_mirage.Server(TCP)
+  module Https = Cohttp_mirage.Server(TLS)
+
+  module Dispatch_http  = Dispatch(C)(DATA)(Http)
+  module Dispatch_https = Dispatch(C)(DATA)(Https)
```
Comment (Drup) [inline static_website_tls/dispatch.ml]: @samoht Someone need to make devices for this pile of functors at some point. :p

**Comment by Drup** [comment]
The whole thing can be shorten a bit using conduit's combinators. They are a bit weird to use. There is an example [here](https://github.com/Drup/No.).

**Comment by azul** [comment]
@drup, I will look into it. Thanks for the pointer!
Can anyone point me to an example of the new logging infrastructure? Would like to include that.

**Comment by talex5** [comment]
I've just updated the [mirage-skeleton/dns](https://github.com/mirage/mirage-skeleton/blob/master/dns/unikernel.ml) example to use Logs. It shows both generating log messages (as a library would do) and configuring a reporter (as an application/unikernel would do).

**Pull request merged**